### DISCUSSION TITLE
Modifying inertia and cg

### DISCUSSION THREAD

**Discussion by fernandafenelon** [JSBSim-Team/jsbsim#582] [Q&A]
Hello everyone, 

I would like to run my model for several CGs and mass' configuration.

Is possible to create a variable in XML that defines the inertia and manipulate it via Python?

![image](https://user-images.githubusercontent.com/96261202/152576137-80c4e29a-8a3f-449a-a931-c1ea9dcd3ecb.png)

**Comment by seanmcleod** [comment]
Currently JSBSim only supports numbers and not properties for the entries you're interested in.

https://github.com/JSBSim-Team/jsbsim/blob/8852bf55a3928a940ff1305f95426695095efacc/src/models/FGMassBalance.h#L83-L112

You could edit your aircraft's xml file and replace the number with a variable name, e.g. `variable_ixx` and `variable_cg_x` and then in your python you could perform a search a replace for these variables by reading your template aircraft xml file and writing out a new aircraft file which you then use each time to start up your simulation run in python.

**Reply by aregis74** [reply]
Hy @seanmcleod, I am seeing your answer about mass property entries and just to make sure if I understood. 
So, there is no way to use something like fdm['inertia/cg-x-in'] = {number} as variable in python?
About the same subject, how jsbsim compute the inertia moments and product of inertia for fuel?
Regards

**Comment by seanmcleod** [comment]
Yep, take a look at the source code and you'll see that all the properties are read-only.

https://github.com/JSBSim-Team/jsbsim/blob/8852bf55a3928a940ff1305f95426695095efacc/src/models/FGMassBalance.cpp#L397-L427

Hence my suggestion that you write some code to do a template match search and replace in your aircraft config file.

One option that does allow you to write/update mass properties is the support for `point masses`.

https://github.com/JSBSim-Team/jsbsim/blob/8852bf55a3928a940ff1305f95426695095efacc/src/models/FGMassBalance.cpp#L431-L448

**Reply by aregis74** [reply]
Ok,  thanks a lot. 
How does jsbsim compute the inertia moments and product of inertia for fuel?

**Comment by seanmcleod** [comment]
> How does jsbsim compute the inertia moments and product of inertia for fuel?

@aregis74 have you taken a look at the source code?
